src/java/picard/analysis/CollectMultipleMetrics.java

```diff
@@ -184,7 +217,10 @@ public int doWork() {
 
         final List<SinglePassSamProgram> programs = new ArrayList<SinglePassSamProgram>();
         for (final ProgramInterface program : new HashSet<ProgramInterface>(programsToRun)) {
-            final SinglePassSamProgram instance = program.makeInstance(OUTPUT , INPUT, REFERENCE_SEQUENCE);
+            if (program.needsReferenceSequence() && REFERENCE_SEQUENCE==null) {
+                throw new PicardException("A program you are trying to run needs a Reference Sequence, please set REFERENCE_SEQUENCE in the command line");
```

Comment:
won't program.toString() do it?



On Thu, Jun 18, 2015 at 11:06 PM, kbergin <notifications@github.com> wrote:

> In src/java/picard/analysis/CollectMultipleMetrics.java
> <https://github.com/broadinstitute/picard/pull/226#discussion_r32772209>:
>
> > @@ -184,7 +217,10 @@ public int doWork() {
> >
> >          final List<SinglePassSamProgram> programs = new ArrayList<SinglePassSamProgram>();
> >          for (final ProgramInterface program : new HashSet<ProgramInterface>(programsToRun)) {
> > -            final SinglePassSamProgram instance = program.makeInstance(OUTPUT , INPUT, REFERENCE_SEQUENCE);
> > +            if (program.needsReferenceSequence() && REFERENCE_SEQUENCE==null) {
> > +                throw new PicardException("A program you are trying to run needs a Reference Sequence, please set REFERENCE_SEQUENCE in the command line");
>
> I was hoping to be able to but couldn't figure it out right away. Let me
> look into it further.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/226/files#r32772209>.
>